aws_lambda_builders/workflows/nodejs_npm/actions.py

```diff
@@ -196,3 +246,122 @@ def execute(self):
+        :type artifacts_dir: str
+        :param artifacts_dir: an existing (writable) directory with project source files.
+            Dependencies will be installed in this directory.
+
+        :type osutils: aws_lambda_builders.workflows.nodejs_npm.utils.OSUtils
+        :param osutils: An instance of OS Utilities for file manipulation
+        """
+
+        super(NodejsNpmLockFileCleanUpAction, self).__init__()
+        self.artifacts_dir = artifacts_dir
+        self.osutils = osutils
+
+    def execute(self):
+        """
+        Runs the action.
+
+        :raises lambda_builders.actions.ActionFailedError: when deleting the lockfile fails
+        """
+
+        try:
+            npmrc_path = self.osutils.joinpath(self.artifacts_dir, "node_modules", ".package-lock.json")
+            if self.osutils.file_exists(npmrc_path):
+                LOG.debug(".package-lock cleanup in: %s", self.artifacts_dir)
+                self.osutils.remove_file(npmrc_path)
+
+        except OSError as ex:
+            raise ActionFailedError(str(ex))
+
+
+class EsbuildBundleAction(BaseAction):
+
+    """
+    A Lambda Builder Action that packages a Node.js package using esbuild into a single file
+    optionally transpiling TypeScript
+    """
+
+    NAME = "EsbuildBundle"
+    DESCRIPTION = "Packaging source using Esbuild"
+    PURPOSE = Purpose.COPY_SOURCE
+
+    def __init__(self, source_dir, artifacts_dir, bundler_config, osutils, subprocess_esbuild):
+        """
+        :type source_dir: str
+        :param source_dir: an existing (readable) directory containing source files
+
+
+        :type artifacts_dir: str
+        :param artifacts_dir: an existing (writable) directory where to store the output.
+            Note that the actual result will be in the 'package' subdirectory here.
+
+        :type osutils: aws_lambda_builders.workflows.nodejs_npm.utils.OSUtils
+        :param osutils: An instance of OS Utilities for file manipulation
+
+        :type subprocess_esbuild: aws_lambda_builders.workflows.nodejs_npm.npm.SubprocessEsbuild
+        :param subprocess_esbuild: An instance of the Esbuild process wrapper
+        """
+        super(EsbuildBundleAction, self).__init__()
+        self.source_dir = source_dir
+        self.artifacts_dir = artifacts_dir
+        self.bundler_config = bundler_config
+        self.osutils = osutils
+        self.subprocess_esbuild = subprocess_esbuild
+
+    def execute(self):
+        """
+        Runs the action.
+
+        :raises lambda_builders.actions.ActionFailedError: when esbuild packaging fails
+        """
+
+        if "entry_points" not in self.bundler_config:
+            raise ActionFailedError("entry_points not set ({})".format(self.bundler_config))
+
+        entry_points = self.bundler_config["entry_points"]
+
+        if not isinstance(entry_points, list):
+            raise ActionFailedError("entry_points must be a list ({})".format(self.bundler_config))
+
+        if len(entry_points) < 1:
```

Comment:
nit: Follow [PEP8](https://www.python.org/dev/peps/pep-0008/) for empty checking.
>For sequences, (strings, lists, tuples), use the fact that empty sequences are false
```
if not entry_points:
```